hbase-balancer/src/main/java/org/apache/hadoop/hbase/CatalogAccessor.java

```diff
@@ -0,0 +1,1071 @@
+import org.apache.hadoop.hbase.util.Pair;
+import org.apache.hadoop.hbase.util.PairOfSameType;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hbase.thirdparty.com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Read/write operations on <code>hbase:meta</code> region as well as assignment information stored
+ * to <code>hbase:meta</code>.
+ * <p/>
+ * Some of the methods of this class take ZooKeeperWatcher as a param. The only reason for this is
+ * when this class is used on client-side (e.g. HBaseAdmin), we want to use short-lived connection
+ * (opened before each operation, closed right after), while when used on HM or HRS (like in
+ * AssignmentManager) we want permanent connection.
+ * <p/>
+ * HBASE-10070 adds a replicaId to HRI, meaning more than one HRI can be defined for the same table
+ * range (table, startKey, endKey). For every range, there will be at least one HRI defined which is
+ * called default replica.
+ * <p/>
+ * <h2>Meta layout</h2> For each table there is single row named for the table with a 'table' column
+ * family. The column family currently has one column in it, the 'state' column:
+ *
+ * <pre>
+ * table:state => contains table state
+ * </pre>
+ *
+ * For the catalog family, see the comments of {@link CatalogFamilyFormat} for more details.
+ * <p/>
+ * TODO: Add rep_barrier for serial replication explanation. See SerialReplicationChecker.
+ * <p/>
+ * The actual layout of meta should be encapsulated inside CatalogAccessor methods, and should not
+ * leak out of it (through Result objects, etc)
+ * @see CatalogFamilyFormat
+ * @see ClientMetaTableAccessor
+ */
+@InterfaceAudience.Private
+public class CatalogAccessor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogAccessor.class);
+  private static final Logger CATALOGLOG =
+    LoggerFactory.getLogger("org.apache.hadoop.hbase.CATALOG");
+
+
+
+  ////////////////////////
+  // Reading operations //
+  ////////////////////////
+
+  /**
+   * Performs a full scan of <code>hbase:meta</code> for regions.
+   * @param connection connection we're using
+   * @param visitor Visitor invoked against each row in regions family.
+   */
+  public static void fullScanRegions(Connection connection, TableName catalogTableName,
+    final ClientMetaTableAccessor.Visitor visitor) throws IOException {
+    scanCatalog(connection, catalogTableName, null, null, QueryType.REGION,
+      null, Integer.MAX_VALUE, visitor);
+  }
+
+  /**
+   * Performs a full scan of <code>hbase:meta</code> for regions.
+   * @param connection connection we're using
+   * @param visitor Visitor invoked against each row in regions family.
+   */
+  public static void fullScanRegions(Connection connection,
+    final ClientMetaTableAccessor.Visitor visitor) throws IOException {
+    scanMeta(connection, null, null, QueryType.REGION, visitor);
```

Comment:
It is scanMeta. I kept the old fullScanRegions() api to avoid having a lot of just renames all over the code. I'm hoping we can do the consolidation as a follow-up? To make this patch a bit more digestible. The same goes for scanMeta().